### ISSUE TITLE
Does not run on Ubuntu 10

### ISSUE THREAD

**Issue by jcoffland** [FoldingAtHome/fah-client-pub#89]
**Reported by @sandstar on 14 Jul 2010 16:33 UTC**
Client errors out on latest ubuntu:

[Platform: linux2 2.6.28-11-server
16:47:41:     Bits: 64
16:47:41:     Mode: Release
16:47:41:************************************ System ************************************
16:47:41:       OS: Linux 2.6.32-23-generic x86_64
16:47:41:   Memory: 752 MB
16:47:41:      CPU: Intel(R) Core(TM)2 Quad CPU Q6600 @ 2.40GHz
16:47:41:   CPU ID: GenuineIntel Family 6 Model 15 Stepping 11
16:47:41:     CPUs: 1 Physical, 2 Logical, 2 Core(s), 1 Thread(s)
16:47:41:  Threads: POSIX_THREADS
16:47:41:     GPUs: 0
16:47:41:********************************************************************************
16:47:41:Enabled computation slot 0: smp:2
FAHClient: relocation error: /lib/libnss_files.so.2: symbol __rawmemchr, version GLIBC_2.2.5 not defined in file libc.so.6 with link time reference
[/code](code]
16:47:41:)

**Comment by jcoffland** [comment]
**Comment by @jcoffland on 14 Jul 2010 20:51 UTC**
This seems to be a Ubuntu problem.  It also affect the v6 SMP clients and other Ubuntu applications.  Here is a solution I found referenced on launchpad:
----
http://foldingforum.org/viewtopic.php?f=58&t=14782

```
sudo apt-get install nscd
# open /etc/nscd.conf with your $EDITOR
# find following line:
# enable-cache hosts no
# change "no" to "yes"
# save & exit the editor
sudo service nscd restart
```

**Comment by jcoffland** [comment]
**Comment by @sandstar on 15 Jul 2010 10:37 UTC**
I'm afraid it is not a fix but a work around. And since nscd is considered broken it's not the perfect work around.

The problem is not limited to Ubuntu but also appears in Fedora and Debian. And probably will occur in every distro that is going to use eglibc.

Note the reply from the debian dev on the bugreport about this issue:
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=589143

**Comment by jcoffland** [comment]
**Comment by @sandstar on 15 Jul 2010 11:41 UTC**
When running the 32bit version on the same machine everthing work fine. Seems that one is compiled differently.

**Comment by jcoffland** [comment]
**Comment by @jcoffland on 15 Jul 2010 20:32 UTC**
OK, I figured this out.  I was previously statically compiling the FAHClient but because of problems like this I switched to what I call '''mostly static''' builds where the build system does some tricks to figure out how to statically link all but a few of the libraries.  However, there was a mistake in the setup for the 64-bit Ubuntu build of the client and it was still being built fully static.  I've fixed this and it should work correctly in the next release.  The 32-bit version worked fine because it was correctly built in '''mostly static''' mode.

Ultimately this is a problem with statically compiled glibc applications that rears it's ugly head on Ubuntu 10.  When it comes right down to it glibc doesn't always work correctly with statically compiled applications especially ones that use networking.

**Comment by jcoffland** [comment]
**Comment by @jcoffland on 20 Jul 2010 17:54 UTC**
Fixed in v7.0.8
